golang/cosmos/vm/client.go

```diff
@@ -34,12 +35,18 @@ var _ rpc.ClientCodec = (*ClientCodec)(nil)
 // having the WriteRequest() method fabricate a Receive() call to clear the rpc
 // state.
 type ClientCodec struct {
-	ctx         context.Context
-	send        func(port, rPort int, msg string)
-	outbound    map[int]rpc.Request
-	inbound     chan *rpc.Response
-	replies     map[uint64]string
+	ctx     context.Context
+	send    func(port, rPort int, msg string)
+	inbound chan *rpc.Response
+
+	reqMutex sync.Mutex
+	outbound map[int]rpc.Request
+
+	// scratch space
 	replyToRead uint64
+
+	mutex   sync.Mutex
+	replies map[uint64]string
```

Comment:
Documenting the scope of these Mutex-valued fields:
```suggestion
	// reqMutex protects outbound
	reqMutex sync.Mutex
	outbound map[int]rpc.Request

	// scratch space
	replyToRead uint64

	// mutex protects replies
	mutex   sync.Mutex
	replies map[uint64]string
```

Reply:
Thanks!  Adopted and expanded.